src/stream/src/executor/aggregation/agg_group.rs

```diff
@@ -185,46 +188,49 @@ impl EstimateSize for AggGroup {
     }
 }
 
-impl AggGroup {
+impl<S: StateStore, Strtg: Strategy> AggGroup<S, Strtg> {
     /// Create [`AggGroup`] for the given [`AggCall`]s and `group_key`.
     /// For [`crate::executor::SimpleAggExecutor`], the `group_key` should be `None`.
     #[allow(clippy::too_many_arguments)]
     pub async fn create(
         group_key: Option<GroupKey>,
         agg_calls: &[AggCall],
-        storages: &[AggStateStorage<impl StateStore>],
-        result_table: &StateTable<impl StateStore>,
+        storages: &[AggStateStorage<S>],
+        result_table: &StateTable<S>,
         pk_indices: &PkIndices,
         row_count_index: usize,
         extreme_cache_size: usize,
         input_schema: &Schema,
-    ) -> StreamExecutorResult<AggGroup> {
+    ) -> StreamExecutorResult<Self> {
         let prev_outputs: Option<OwnedRow> = result_table
             .get_row(group_key.as_ref().map(GroupKey::table_pk))
             .await?;
         if let Some(prev_outputs) = &prev_outputs {
             assert_eq!(prev_outputs.len(), agg_calls.len());
         }
 
-        let states =
-            futures::future::try_join_all(agg_calls.iter().enumerate().map(|(idx, agg_call)| {
-                AggState::create(
-                    agg_call,
-                    &storages[idx],
-                    prev_outputs.as_ref().map(|outputs| &outputs[idx]),
-                    pk_indices,
-                    group_key.as_ref(),
-                    extreme_cache_size,
-                    input_schema,
-                )
-            }))
+        // XXX(wrj): don't use try_join_all here to avoid weird memmove overhead
+        let mut states = Vec::with_capacity(agg_calls.len());
+        for (idx, agg_call) in agg_calls.iter().enumerate() {
+            let state = AggState::create(
+                agg_call,
+                &storages[idx],
+                prev_outputs.as_ref().map(|outputs| &outputs[idx]),
+                pk_indices,
+                group_key.as_ref(),
+                extreme_cache_size,
+                input_schema,
+            )
```

Comment:
This can potentially affect performance of initialization/recovery if later introduced UDAF need to do IO on init?

Reply:
Yes. This can be revert after #11275.